### PR TITLE
Make AdaLAM output match confidence

### PR DESCRIPTION
**Pull request by ducha-aiki**
#### Changes
Based on https://github.com/kornia/kornia/pull/1831#discussion_r962703620

#### Type of change

- [x] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🔬 New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: kornia/feature/adalam/adalam.py [hunk 1]
```diff
@@ -151,6 +152,7 @@ def filter_matches(
             s1/s2: keypoint scales. They can be None if 'scale_rate_threshold' in config is set to None.
                    See documentation on 'scale_rate_threshold' in the DEFAULT_CONFIG.
                    Expected a float32 tensor with shape (num_keypoints_in_source/destination_image,)
+            return_dist: if True, inverse confidence value is also outputed. Default is False
```
Comment (edgarriba) [inline kornia/feature/adalam/adalam.py:155]:
```suggestion
            return_dist: if True, inverse confidence value is also outputted.
```

File: kornia/feature/adalam/adalam.py [hunk 2]
```diff
@@ -200,6 +216,7 @@ def match_and_filter(self, k1, k2, d1, d2, im1shape=None, im2shape=None, o1=None
             s1/s2: keypoint scales. They can be None if 'scale_rate_threshold' in config is set to None.
                    See documentation on 'scale_rate_threshold' in the DEFAULT_CONFIG.
                    Expected an array with shape (num_keypoints_in_source/destination_image,)
+            return_dist: if True, inverse confidence value is also outputed. Default is False
```
Comment (edgarriba) [inline kornia/feature/adalam/adalam.py:219]:
```suggestion
            return_dist: if True, inverse confidence value is also outputted.
```

File: kornia/feature/adalam/core.py [hunk 3]
```diff
@@ -304,13 +306,23 @@ def adalam_core(
 
     conf = inl_confidence[ransidx[inlier_idx]]
     cnt = inlier_counts[ransidx[inlier_idx]].float()
-    passed_inliers_mask = (conf >= MIN_CONF) & (cnt * (1 - 1 / conf) >= MIN_INLIERS)
+    dist_ratio = 1.0 / conf
+    passed_inliers_mask = (conf >= MIN_CONF) & (cnt * (1 - dist_ratio) >= MIN_INLIERS)
```
Comment (edgarriba) [inline kornia/feature/adalam/core.py:310]: not sure if is safer to user logical_and here

File: kornia/feature/adalam/core.py [hunk 4]
```diff
@@ -304,13 +306,23 @@ def adalam_core(
...
     if final_matches.shape[0] > 1:
-        return torch.unique(final_matches, dim=0)
+        # https://stackoverflow.com/a/72005790
+        final_matches, idxs, counts = torch.unique(final_matches, dim=0, return_inverse=True, return_counts=True)
+        _, ind_sorted = torch.sort(idxs)
+        cum_sum = counts.cumsum(0)
+        cum_sum = torch.cat((torch.tensor([0]), cum_sum[:-1]))
+        first_indicies = ind_sorted[cum_sum]
+        accepted_dist = accepted_dist[first_indicies]
+    if return_dist:
```
[the remark this answers is not in the document]
Comment (ducha-aiki) [reply kornia/feature/adalam/core.py:326]: I was thinking about AdaLAM users, which use it as is, for backwards compatability. 
For the new kornia function `match_adalam` always outputs distance.

**Pull request merged**